R/plot_jags_tracedx.R

```diff
@@ -34,50 +37,59 @@
 plot_jags_trace <- function(data,
                             iso = unique(data$Iso_type),
                             param = unique(data$Parameter),
-                            strat = unique(data$Stratification)) {
+                            strat = unique(data$Stratification),
+                            id = c("newperson")) {
 
   attributes_jags <- data[["attributes"]]
-
-  trace_strat_list <- list()
-  for (i in strat) {
+  
+  trace_id_list <- list()
+  for (h in id) {
 
     visualize_jags_sub <- data |>
-      dplyr::filter(.data$Stratification == i) |>
-      dplyr::filter(.data$Subject == "newperson")
+      dplyr::filter(.data$Subject == h)
+    
+    trace_strat_list <- list()
+  
+    for (i in strat) {
+
+      visualize_jags_sub <- data |>
+        dplyr::filter(.data$Stratification == i)
```

Comment:
It looks like we start from the full `data` again inside the strat loop, so the earlier `Subject == h` filter can be lost. If we continue filtering from the already-subset data, the id argument will always be respected.

```suggestion
visualize_jags_sub <- visualize_jags_sub |>
  dplyr::filter(.data$Stratification == i)
```